src/figuresClasses.ts

```diff
@@ -1,19 +1,95 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  shape: Shape;
+  color: Color;
 
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shape;
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.shape = 'triangle';
+    this.color = color;
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Side length should be more than 0');
+    }
+
+    const longestSide = Math.max(a, b, c);
+    const sum = a + b + c;
 
-export class Circle {
+    if (longestSide >= sum - longestSide) {
+      throw Error('The side cannot be more than a sum of two others');
+    }
+  }
 
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
```

Comment:
consider to make more descriptrive variables
```suggestion
    const halfPerimeter = (this.a + this.b + this.c) / 2;
```